cpu/cc3200/periph/gpio.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (C) 2019 Ludwig Maximilian Universität
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup        cpu_cc3200
+ * @{
+ *
+ * @file
+ * @brief           GPIO driver implementation
+ *
+ * @author          Wladislaw Meixner <wladislaw.meixner@campus.lmu.de>
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "cpu.h"
+#include "periph/gpio.h"
+#include "vendor/hw_gpio.h"
+#include "vendor/hw_memmap.h"
+#include "vendor/hw_ocp_shared.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifdef MODULE_PERIPH_GPIO_IRQ
+/**
+ * @brief   static callback memory
+ */
+static gpio_isr_ctx_t isr_ctx[4][8];
+#endif /* MODULE_PERIPH_GPIO_IRQ */
+
+#define GPIO_PINS_PER_PORT 8     /**< Number of pins per port */
+#define GPIO_DIR_MASK 0x00000001 /**< GPIO direction configuration mask */
+#define PIN_MODE_GPIO \
+    0x00000000 /**< GPIO Pin type value used to configure pin to GPIO */
+#define PAD_MODE_MASK 0x0000000F /**< GPIO mask for setting pin mode */
+#define PAD_STRENGTH_MASK                              \
+    0x000000E0 /**< GPIO mask for setting pin strength \
+                */
+#define PAD_TYPE_MASK                                        \
+    0x00000310 /**< GPIO mask for setting pin type e.g. PULL \
+                */
+#define PAD_CONFIG_BASE ((OCP_SHARED_BASE + OCP_SHARED_O_GPIO_PAD_CONFIG_0))
+
+/**
+ * @brief get hardware configuration register for a pin
+ *
+ */
+#define PAD_CONFIG_REG(pin) \
+    (*((volatile unsigned long *)((gpio_pin << 2) + PAD_CONFIG_BASE)))
+
+/**
+ * @brief gpio base addresses
+ *
+ */
+static unsigned long ports[] = { GPIOA0_BASE, GPIOA1_BASE, GPIOA2_BASE,
+                                 GPIOA3_BASE, GPIOA4_BASE };
+
+/**
+ * @brief pin to GPIO pin numbers mappings
+ *
+ */
+static const unsigned long pin_to_gpio_num[64] = {
```

Comment:
Does this have to be an `unsigned long`? All the values fit into an `uint8_t`.

Reply:
That definitely can be done here. Should save a nice chunk of memory!